File: mobile/src/main/java/net/mm2d/android/upnp/cds/chapter/FetcherFactory.java
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.android.upnp.cds.chapter;

import net.mm2d.android.upnp.cds.CdsObject;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import io.reactivex.Single;

/**
 * @author <a href="mailto:ryo@mm2d.net">大前良介 (OHMAE Ryosuke)</a>
 */
interface FetcherFactory {
    @Nullable
    Single<List<Integer>> create(@NonNull CdsObject object);
}

File: mobile/src/main/java/net/mm2d/android/upnp/cds/chapter/FetcherFactory.kt
/*
 * Copyright (c) 2017 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.android.upnp.cds.chapter

import io.reactivex.Single
import net.mm2d.android.upnp.cds.CdsObject

/**
 * @author [大前良介 (OHMAE Ryosuke)](mailto:ryo@mm2d.net)
 */
internal interface FetcherFactory {
    fun create(
        cdsObject: CdsObject,
    ): Single<List<Int>>?
}
